### ISSUE TITLE
baseline test plan

### ISSUE THREAD

**Issue by raulk** [ipfs/testground#597]
I'd like a test plan that I can run with any arbitrary number of nodes that exercises ONLY the testground components. We should exercise:

1. Network initialised.
2. Configuring the network. Waiting for a callback.
3. Signal entry on a barrier.
4. Wait on barrier on 20% nodes, barrier on 40% nodes, barrier on 60%, barrier on 80%, barrier on 100% of nodes.
5. Publish an item on a subtree.
6. Subscribe to the subtree.

Print time elapsed:

1. Between the time the run was scheduled and the task was started (for this, we need to inject a "run start" time in an env variable?).
2. Until network is initialised.
3. Until network is reconfigured.
4. Time to hit the barrier at the above checkpoints.
5. Time to publish the item on the subtree.
6. Time until we received 20%, 40%, 60%, 80%, 100% of items in the subtree via the subscription.

We should be able to visualise all these metrics in Grafana.

**Comment by coryschwartz** [comment]
The baseline benchmarks I added in 604 did *most* of the things requested here.
I have added the prometheus metrics to the SDK, and then integrated the new metrics SDK to the baseline tests in https://github.com/ipfs/testground/pull/640/. The grafana dashboard is added there as well... although the dashboard does need some work. It's a bit too basic right now.

The last thing I need to do for this is to test publishing items onto a subtree and time the subscription.
